get-resource-index

```diff
@@ -1,14 +1,44 @@
 #!/bin/bash
 
 base="$(dirname "$0")"
-resource_index=$(jq -r '.RESOURCE_INDEX' "$base"/env/production/config.json)
 
-if [[ -n "$RESOURCE_INDEX_PREFIX" ]]; then
-    s3path="${resource_index#s3://}"
-    filename=$(basename "$s3path")
+main() {
+    local jq resource_index s3path filename new_filename
 
-    new_filename="tmp/$RESOURCE_INDEX_PREFIX/$filename"
-    resource_index="${resource_index/$filename/$new_filename}"
-fi
+    if ! jq="$(command -v jq 2>/dev/null)"; then
+        jq="$(download)"
+    fi
 
-echo "$resource_index"
+    resource_index=$("$jq" -r '.RESOURCE_INDEX' "$base"/env/production/config.json)
+
+    if [[ -n "$RESOURCE_INDEX_PREFIX" ]]; then
+        s3path="${resource_index#s3://}"
+        filename=$(basename "$s3path")
+
+        new_filename="tmp/$RESOURCE_INDEX_PREFIX/$filename"
+        resource_index="${resource_index/$filename/$new_filename}"
+    fi
+
+    echo "$resource_index"
+}
+
+download() {
+    local path system_name platform
+
+    path="$(mktemp -t "jq.XXXXXX")"
+    system_name="$(uname -s | tr A-Z a-z)"
+
+    if [[ "$system_name" == "darwin" ]]; then
+        platform="osx-amd64"
+    else
+        platform="linux64"
+    fi
+
+    curl -fsSL --proto '=https' "https://github.com/jqlang/jq/releases/download/jq-1.6/jq-$platform" \
```

Comment:
Instead of downloading `jq` because it's not available in our Heroku environment, why not rewrite `get-resource-index` as a node script and utilise the native JSON parsing in node?

Reply:
That was going to be my next step if my latest changes didn't work, but the Heroku logs show the app using the correct tmp resource index now:

```
2024-05-01T00:14:11.886398+00:00 app[web.1]: [verbose]  [RESOURCE INDEX UPDATE S3] Fetching index from s3://nextstrain-inventories/resources/tmp/separate-indexer/v1.json.gz
```

---
I can still rewrite it as a node script if that's preferred, jq was just simpler for me than a node script 😅

Reply:
I would prefer it to be in node. 

Could we also move it into `/scripts/`?